roles/ceph-defaults/tasks/facts.yml

```diff
@@ -74,7 +75,7 @@
   set_fact:
     fsid: "{{ ceph_current_fsid.stdout }}"
   when:
-    - ceph_current_fsid.rc == 0
+    - ceph_current_fsid.get('rc', 1) == 0
```

Comment:
This seems a bit weird, it's basically defaulting to 1 if ceph_current_fsid.rc doesn't exist?
Since we're using run_once above wouldn't that mean only the first host has "ceph_current_fsid" set, and as a result all other hosts would adopt "1".

Reply:
@andymcc no, see below 

```
---
- name: run /bin/true
  command: 'true'
  register: bin_true
  run_once: true

- name: print bin_true
  debug:
    msg: "{{ bin_true.get('rc', 1) }}"
```

```
TASK [osd : run /bin/true] *****************************************************************************************************************************************************************************************************************************************************
task path: /Users/guits/GIT/playbook-test/roles/osd/tasks/main.yml:2
changed: [osd0] => {"changed": true, "cmd": ["true"], "delta": "0:00:00.007441", "end": "2018-04-12 14:04:08.639327", "rc": 0, "start": "2018-04-12 14:04:08.631886", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [osd : print bin_true] ****************************************************************************************************************************************************************************************************************************************************
task path: /Users/guits/GIT/playbook-test/roles/osd/tasks/main.yml:7
ok: [osd0] => {
    "msg": "0"
}
ok: [osd1] => {
    "msg": "0"
}
ok: [osd2] => {
    "msg": "0"
}
```

Reply:
ahh interesting, so setting the fact on the first host causes it to be accessible to all others?